kernel: CONFIG_INIT_STACKS : minor documentation & dependency update

While investigating #7128 , I looked into the k-config option for initializing stacks:

> 
> config INIT_STACKS
> 	bool
> 	prompt "Initialize stack areas"
> 	default n
> 	help
> 	  This option instructs the kernel to initialize stack areas with a
> 	  known value (0xaa) before they are first used, so that the high
> 	  water mark can be easily determined. This applies to the stack areas
> 	  for threads.

At least for ARM, CONFIG_INIT_STACKS initializes the _interrupt stack_, as well, so the last sentence is not accurate, IMHO.

In addition to the above: `CONFIG_INIT_STACKS=y ` looks to me that it has a dependency on `CONFIG_MULTITHREADING=y `(attempt to define the first and not the latter leads to compile error: undefined reference to `_impl_k_thread_create` [used in k_thread_create]). I think this dependency should be reflected in the dependencies of `config INIT_STACKS`, to prevent accidental enabling of CONFIG_INIT_STACKS

--- a/arch/arm/core/thread.c
+++ b/arch/arm/core/thread.c
@@ -15,9 +15,6 @@
 #include <toolchain.h>
 #include <kernel_structs.h>
 #include <wait_q.h>
-#ifdef CONFIG_INIT_STACKS
-#include <string.h>
-#endif /* CONFIG_INIT_STACKS */
 
 #ifdef CONFIG_USERSPACE
 extern u8_t *_k_priv_stack_find(void *obj);

--- a/kernel/Kconfig
+++ b/kernel/Kconfig
@@ -283,7 +283,7 @@ config INIT_STACKS
 	  This option instructs the kernel to initialize stack areas with a
 	  known value (0xaa) before they are first used, so that the high
 	  water mark can be easily determined. This applies to the stack areas
-	  for threads.
+	  for threads, as well as to the interrupt stack.
 
 config KERNEL_DEBUG
 	bool "Kernel debugging"

